### PR TITLE
[INLONG-3550][Manager] Support File source in client and server

### LINKED ISSUE #3550: [Improve][Manager] Add file source in Manager Client
### Description

_No response_

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by kipshi**
### Title Name: [INLONG-3550] Support File source in client and server

where *XYZ* should be replaced by the actual issue number.

Fixes #3550 

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/source/file/FileSourceDTO.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    @ApiModelProperty("Path regex pattern for file, such as /a/b/*.txt")
+    private String pattern;
+
+    @ApiModelProperty("TimeOffset for collection, "
+            + "'1m' means one minute before, '1h' means one hour before, '1d' means one day before, "
+            + "Null means from current timestamp")
+    private String timeOffset;
+
+    @ApiModelProperty("Addition attributes for file source, save as a=b&c=d&e=f ")
+    private String additionalAttr;
```
Comment (healchow) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/source/file/FileSourceDTO.java:57]: Is it better to name this parameter `extParams`? Keep the same naming as other places.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java [hunk 2]
```diff
@@ -175,6 +174,15 @@ public TaskResult getTaskResult(TaskRequest request) {
         List<StreamSourceEntity> needIssuedList = sourceMapper.selectByStatusAndIp(statusList, agentIp, uuid);
         entityList.addAll(needIssuedList);
 
+        List<StreamSourceEntity> fileEntityList = sourceMapper.selectByStatusAndType(needAddStatusList,
+                Lists.newArrayList(SourceType.FILE.getType()), TASK_FETCH_SIZE * 2);
+        for (StreamSourceEntity fileEntity : fileEntityList) {
+            FileSourceDTO fileSourceDTO = FileSourceDTO.getFromJson(fileEntity.getExtParams());
+            if (agentIp.equals(fileSourceDTO.getIp())) {
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java:181]: Why not put the `agentIp` into the `selectByStatusAndType` method to filter result?

Comment (kipshi) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java:181]: 1 agentIp is empty if not fetched by agent
2 selectByType is useful sooner or later

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/source/file/FileSourceDTO.java (hunk 1 above)
Comment (kipshi) [reply inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/source/file/FileSourceDTO.java:57]: additionalAttr is used by agent, not by manager

File: inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/pojo/FileJob.java [hunk 3]
```diff
@@ -54,12 +51,11 @@
     @Data
     public static class FileJobTaskConfig {
 
-        private String dataName;
-        private String path;
-        private int taskId;
         private String pattern;
-        private String cycleUnit;
+        //'1m' means one minute before, '1h' means one hour before, '1d' means one day before
```
Comment (EMsnap) [inline inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/pojo/FileJob.java:55]: 1m means one minute after, -1m means one minute before

**Review by healchow — APPROVED** [review]

**Review by EMsnap — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**